### PR TITLE
MSVC changes to build gnustep-gui

### PR DESCRIPTION
**Pull request by gcasa**
Complete libs-gui support in this toolkit

### CODE REVIEW

**Comment by triplef** [comment]
Would be great if you could rebase this so we get the fixes for the CI builds from master.

**Comment by gcasa** [comment]
> Would be great if you could rebase this so we get the fixes for the CI builds from master.

It seems it is STILL FAILING....

```
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(43,7): warning: forward references to 'enum' types are a Microsoft extension [-Wmicrosoft-enum-forward-reference]
        enum FooManChu enumDefault;
             ^
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(73,11): error: unknown type name 'atomic_bool'
@property atomic_bool atomicBoolDefault;
          ^
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(154,11): error: unknown type name 'atomic_bool'
@property atomic_bool atomicBoolDefault;
          ^
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(195,2): error: unknown type name 'atomic_bool'
        atomic_bool atomicBoolDefault;
        ^
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(493,12): warning: cast to smaller integer type 'int' from 'id' [-Wpointer-to-int-cast]
    return (int)object_getIvar(self, ivar);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(701,18): error: assignment to readonly property
    t.intDefault = 2;
    ~~~~~~~~~~~~ ^ ~
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(708,21): error: assignment to readonly property
    t.structDefault = struct1;
    ~~~~~~~~~~~~~~~ ^ ~~~~~~~
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(726,16): error: assignment to readonly property
    t.idRetain = testValue;
    ~~~~~~~~~~ ^ ~~~~~~~~~
D:\a\tools-windows-msvc\tools-windows-msvc\src\libobjc2\Test\PropertyIntrospectionTest2_arc.m(736,16): error: assignment to readonly property
    t.idRetain = testValue2;
    ~~~~~~~~~~ ^ ~~~~~~~~~~
3 warnings and 13 errors generated.
[111/223] Linking C executable Test\PropertyIntrospectionTest_optimised.exe
ninja: build stopped: subcommand failed.
Failed 
Error: Process completed with exit code 1.
```

**Comment by triplef** [comment]
That’s because you overwrote `.github/workflows/ci.yml`.

File: patches/gnustep-gui-private.patch [hunk 1]
```diff
@@ -831,19 +831,10 @@ index d3bf78e6b..052ed12c8 100644
  + (NSGraphicsContext *) graphicsContextWithAttributes: (NSDictionary *)attributes;
```
Comment (triplef) [inline patches/gnustep-gui-private.patch:831]: Wouldn’t it be better to set up an windows-msvc branch for libs-gui with all your changes? That way they could already be run through the CI as well to make sure nothing else breaks, and the build script here could just point to that branch.

Comment (gcasa) [reply patches/gnustep-gui-private.patch:831]: I would rather have patches against master instead of some arbitrary branch

Comment (gcasa) [reply patches/gnustep-gui-private.patch:831]: Also, some of these changes should make it into master.

Comment (triplef) [reply patches/gnustep-gui-private.patch:831]: Hopefully all of them! (Conditionally if needed.)

Comment (gcasa) [reply patches/gnustep-gui-private.patch:831]: Is there a define that is set by msvc that we can use in some cases to conditionally compile things?   If there isn't one we should define one.

Comment (triplef) [reply patches/gnustep-gui-private.patch:831]: Yeah, the Microsoft compiler defines `_MSC_VER`. See e.g. here:
https://github.com/gnustep/libs-base/blob/6657796e1599d57410a6e837005216232e9f440a/Source/NSFileManager.m#L78

**Comment by gcasa** [comment]
I believe I can simply close this as @qmfrederik has incorporated these changes into master. @fredkiefer @triplef

**Pull request closed without merging**